app/src/main/java/com/andrew/tasky/agenda/data/util/DateTimeConversionImpl.kt

```diff
@@ -0,0 +1,25 @@
+package com.andrew.tasky.agenda.data.util
+
+import android.icu.util.TimeZone
+import com.andrew.tasky.agenda.domain.DateTimeConversion
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZonedDateTime
+
+class DateTimeConversionImpl : DateTimeConversion {
+
+    override fun localDateTimeToZonedEpochMilli(localDateTime: LocalDateTime): Long {
+        return ZonedDateTime.of(
+            localDateTime,
+            ZoneId.of(TimeZone.getDefault().id)
+        ).toInstant().toEpochMilli()
+    }
+
+    override fun zonedEpochMilliToLocalDateTime(zonedEpochMilli: Long): LocalDateTime {
+        return ZonedDateTime.ofInstant(
+            Instant.ofEpochMilli(zonedEpochMilli),
+            ZoneId.of(TimeZone.getDefault().id)
+        ).toLocalDateTime()
+    }
+}
```

Comment:
For this you don't need to put the implementation in data, since there's nothing Android related about `ZonedDateTime`. If something comes from Android or just Java/Kotlin, you can easily see at the imports. However, what would make this Android related is if you use the Android specific `TimeZone` import (there are two - one from java and one from Android).
The difference is that the one from Android will consider the time zone the user set in their Android settings which is why I'd recommend that one for apps, otherwise it might lead to incorrect times. In that case, I'd only put that time zone specific implementation in data, but leave the conversion up to domain. For example like this:
```
// domain
interface TimeZoneProvider {
    fun getTimeZone(timeZoneId: String?): ZoneId
}

// data
import android.icu.util.TimeZone

object AndroidTimeZoneProvider : TimeZoneProvider {
    override fun getTimeZone(timeZoneId: String?): ZoneId {
        // Check if zone ID is supported
        val adjustedTimeZone = if (timeZoneId in TimeZone.getAvailableIDs()) {
            timeZoneId
        } else TimeZone.getDefault().id // if not, fallback to default time zone

        return ZoneId.of(adjustedTimeZone)
    }
}

// Time zone conversion in domain
class DateTimeConversion(private val timeZoneProvider: TimeZoneProvider) {
    
    // Your conversion functions here
} 
```